examples/c/src/phi3_terminate.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <cassert>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include "ort_genai.h"
+
+#include <thread>
+#include <vector>
+
+// C++ API Example for Terminating Session
+
+void Generate_Output_CXX(OgaGenerator* generator, std::unique_ptr<OgaTokenizerStream> tokenizer_stream) {
+  try {
+    while (!generator->IsDone()) {
+      generator->ComputeLogits();
+
+      generator->GenerateNextToken();
+
+      std::unique_ptr<OgaTensor> output_logits = generator->GetOutput("logits");
+
+      auto logits = reinterpret_cast<float*>(output_logits->Data());
+
+      const auto num_tokens = generator->GetSequenceCount(0);
+      const auto new_token = generator->GetSequenceData(0)[num_tokens - 1];
+      std::cout << tokenizer_stream->Decode(new_token) << std::flush;
+    }
+  }
+  catch (const std::exception& e) {
+    std::cout << "Session Terminated: " << e.what() << std::endl;
+  }
+}
+
+void Generator_SetTerminate_Call(OgaGenerator* generator) {
+  std::cout << "Calling SetTerminate for Phi3 example after 1 second" << std::endl;
+  std::this_thread::sleep_for(std::chrono::seconds(1));
+  generator->SetRuntimeOption("terminate_session", "1");
+}
+
+
+void CXX_API(const char* model_path) {
+  std::cout << "Creating model..." << std::endl;
+  auto model = OgaModel::Create(model_path);
+  std::cout << "Creating tokenizer..." << std::endl;
+  auto tokenizer = OgaTokenizer::Create(*model);
+  
+
+  bool runTerminate = true;
+
+  try {
+    while (true) {
+      auto tokenizer_stream = OgaTokenizerStream::Create(*tokenizer);
+      std::string text;
+      std::cout << "Prompt: (Use quit() to exit)" << std::endl;
+      std::getline(std::cin, text);
+
+      if (text == "quit()") {
+        break;  // Exit the loop
+      }
+
+      const std::string prompt = "<|user|>\n" + text + "<|end|>\n<|assistant|>";
+
+      auto sequences = OgaSequences::Create();
+      tokenizer->Encode(prompt.c_str(), *sequences);
+
+      std::cout << "Generating response..." << std::endl;
+      auto params = OgaGeneratorParams::Create(*model);
+      params->SetSearchOption("max_length", 1024);
+      params->SetInputSequences(*sequences);
+
+      auto generator = OgaGenerator::Create(*model, *params);
+
+      std::vector<std::thread> threads;
```

Comment:
Are there only just two threads?

Reply:
Yes, in this example there are only two threads.